### PR TITLE
Joshfischer/asf site

### PR DESCRIPTION
**Pull request by joshfischer1108**
@nwangtw This is a first pass on a script to build and publish our static site.  I borrowed a lot of the steps from the incubator site build process here -> https://github.com/apache/incubator/blob/master/build_site.sh.  I've turned off the ci  job for now.  Will turn it on once we come to agreement on how the script should work.

Any feedback is welcome.

### CODE REVIEW

**Comment by joshfischer1108** [comment]
My thinking is that we will build off of master at some time period we decide on, similar to what the incubator site does.  We can then take the generated static files out of the `website/public` directory and move them to a folder called `content` in the root directory.  These changes can be committed to the `asf-site` branch.  We can then use apache infra to serve the files out of the `content` directory.  Thoughts?

**Comment by nwangtw** [comment]
Currently the website is single version only. I feel eventually we will need to have web pages for each version. Therefore I am thinking that we can have a ci to make sure website is in good shape, and we should add it to our release process (generate website after a release vote is successful). Just my 2 cents.

File: website/build_site.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
```
Comment (nwangtw) [inline website/build_site.sh]: License header.

File: website/build_site.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+rm -rf $WORK_DIR
+mkdir -p $WORK_DIR
+
+cd website
+
+make setup
+
+make site
+
+# push all of the results to asf-site branch
```
Comment (nwangtw) [inline website/build_site.sh]: Maybe breaking the code below into a publish_website.sh could be more flexible.

File: website/build_site.sh (hunk 1 above)
Comment (joshfischer1108) [reply website/build_site.sh]: Ahh yes. Thanks

File: website/build_site.sh (hunk 2 above)
Comment (joshfischer1108) [reply website/build_site.sh]: I agree

**Comment by joshfischer1108** [comment]
Is the release process manual right now?  Might be a good time to automate that while we are working on the website deployment.

**Comment by nwangtw** [comment]
> Is the release process manual right now? Might be a good time to automate that while we are working on the website deployment.

There are jenkins jobs for building artifacts. Last three in this page: https://builds.apache.org/search/?q=heron&Jenkins-Crumb=edf72a5514450f709b23ec4c0f42f56a.

**Comment by nwangtw** [comment]
> Is the release process manual right now? Might be a good time to automate that while we are working on the website deployment.

And the release process itself is manual currently after artifacts are built.

**Comment by nwangtw** [comment]
LGTM~

**Pull request merged**